Multisig/Logic/Models/KeyInfo.swift

```diff
@@ -393,6 +393,34 @@ extension KeyInfo {
         if let key = (try? privateKey()) { return key }
         return try? delegatePrivateKey()
     }
+
+    func pushNotificationSigningKey(completion: @escaping (Result<PrivateKey?, Error>) -> ()) {
+        privateKey { [unowned self] result in
+            switch result {
+            case .success(let privateKey):
+                if privateKey == nil {
+                    delegatePrivateKey(completion: completion)
+                } else {
+                    completion(.success(privateKey))
+                }
+            case .failure(let error):
+                completion(.failure(error))
+            }
+        }
+    }
+
+    func pushNotificationSigningKey() async throws -> PrivateKey? {
```

Comment:
So this method is like a wrapper for the previous one, I don't see the why we need it actually, I would just keep the previous one and use it directly

Reply:
getting delegate key from keystore is async operation